File: POO_2026_1/src/POO/Time.java
package POO;

import java.util.ArrayList;

public class Time {
  private int idTime;
  private String nomeTime;
  private Pessoa tecnico; // relacionamento de associação
  private ArrayList<Jogador> jogadores; // relacionamento de agregação

    public Time(int idTime, String nomeTime, Pessoa tecnico) {
        this.idTime = idTime;
        this.nomeTime = nomeTime;
        this.tecnico = tecnico;
        
        this.jogadores = new ArrayList();
    }

    public int getIdTime() {
        return idTime;
    }

    public void setIdTime(int idTime) {
        this.idTime = idTime;
    }

    public String getNomeTime() {
        return nomeTime;
    }

    public void setNomeTime(String nomeTime) {
        this.nomeTime = nomeTime;
    }

    public Pessoa getTecnico() {
        return tecnico;
    }

    public void setTecnico(Pessoa tecnico) {
        this.tecnico = tecnico;
    }
    
    public void adicionarJogador(Jogador novoJogador) {
      jogadores.add(novoJogador);
    }
    
    public void mostrarTime() {
      System.out.printf("%s\n", nomeTime);
      System.out.printf("Técnico: %s (%d anos)\n\n",
        tecnico.getNome(), tecnico.getDtNasc().calcularIdade());
      
      Jogador.listarJogadores(jogadores);
    }

}

File: CPP/POO_CPP/TimeDeFutebol_CPP/Time.cpp
#include <iostream>
#include "Time.h"
#include "C:\Meus Documentos\Professor do DAELN\Aulas\TecProg\CPP\POO_CPP\CadastrarPessoa_CPP\Pessoa.h"
#include "C:\Meus Documentos\Professor do DAELN\Aulas\TecProg\CPP\POO_CPP\JogadoresDeFutebol_CPP\Jogador.h"

using namespace std;

Time::Time(int idTime, string nomeTime, Pessoa *tecnico) {
  this->idTime = idTime;
  this->nomeTime = nomeTime;
  this->tecnico = tecnico;
}

Time::~Time() {
}

int Time::getIdTime() {
  return idTime;
}

void Time::setIdTime(int idTime) {
  this->idTime = idTime;
}

string Time::getNomeTime() {
  return nomeTime;
}

void Time::setNomeTime(string nomeTime) {
  this->nomeTime = nomeTime;
}

Pessoa *Time::getTecnico() {
  return tecnico;
}

void Time::setTecnico(Pessoa *tecnico) {
  this->tecnico = tecnico;
}

void Time::adicionarJogador(Jogador *novoJogador) {
  jogadores.push_back(novoJogador);
}

void Time::mostrarTime() {
  char linha[100];

  cout << getNomeTime() << endl << endl;
  cout << "Técnico: " << getTecnico()->getNome() << endl << endl;
  cout << "Nome do Jogador............... .Idade. Posição............. ...Salário..." << endl;
  cout << "=========================================================================" << endl;

  double totSal = 0;
  for(Jogador *item: jogadores) {
    cout << item->toString();
    totSal = totSal + item->getSalario();
  }

  cout << "=========================================================================" << endl;
  sprintf(linha, "%60s R$ %10.2f\n", "TOTAL SALÁRIOS", totSal);
  cout << linha;
}
